### PR TITLE
Fix HTTP2 error with openshift routes

### PR DESCRIPTION
**Pull request by sayan-biswas**
This is a workaround for the bug [PLNSRVCE-1210](https://issues.redhat.com/browse/PLNSRVCE-1210)

This error is appearing in the API service logs because when we are creating a route in an Openshift cluster with HTTP2 enabled, HAPROXY opens  TCP connection to that route for health check, and maybe closes the connection before the handshake is complete. The `net` golang package logs these requests.
But I have observed that these issue doesn't happen if we separate the ports in different service (not sure why this behaviour)

### CODE REVIEW

**Review by gabemontero — COMMENTED** [review]
@sayan-biswas were the repeating http2 logs showing up in the api server instance running during our CI?  In other words, is enough of the HAPROXY setup done in our CI to reproduce this?

If the answer is "yes" to any of that, it would be nice to update our tests here to dump the api server log and confirm the repeating error is no longer occurring.

**Review by Roming22 — APPROVED** [review]

**Review by ramessesii2 — COMMENTED** [review]
lgtm. I'll wait for Sayan's reply on Gabe's ask.

**Comment by sayan-biswas** [comment]
> @sayan-biswas were the repeating http2 logs showing up in the api server instance running during our CI? In other words, is enough of the HAPROXY setup done in our CI to reproduce this?
> 
> If the answer is "yes" to any of that, it would be nice to update our tests here to dump the api server log and confirm the repeating error is no longer occurring.

No, it doesn't. Probably that's why we have not noticed. This was reported by Adam as it was prominent in staging and prod setup. However to get this reproducible in our dev setup we need to enable the HTTP2, which is not enabled by default.

`oc annotate ingresses.config/cluster ingress.operator.openshift.io/default-enable-http2=true`

I'll add this in the dev setup script. Also I'll add some test if possible.

File: operator/test/test.sh [hunk 1]
```diff
@@ -288,6 +288,13 @@ test_security() {
 }
 
 test_results() {
+  # Check logs for OCP bug https://issues.redhat.com/browse/OCPBUGS-5916
+  pattern="http2: server: error reading preface from client +.+ read: connection reset by peer"
+  if kubectl logs deployment/tekton-results-api -c "api" -n "$NAMESPACE" 2>&1 | grep -ciqE "$pattern"; then
```
Comment (gabemontero) [inline operator/test/test.sh:293]: I assume this `grep` is right, but I have no idea :-) 

You got a quick sample to confirm @sayan-biswas ?

Or, was there a reason checking for `http2: server: error reading preface from client` was not sufficient ?

**Review by gabemontero — COMMENTED** [review]
Just a minor question @sayan-biswas otherwise looks great thanks!

@Roming22 FYI

File: operator/test/test.sh (hunk 1 above)
Comment (sayan-biswas) [reply operator/test/test.sh:293]: Ya, its right, tested it. I used as a regular expression.
But I'll update and just put `http2: server: error reading preface from client` instead, to catch broader errors. I guess the rest of the messages are unwrapped errors from other methods down the stack.
I added few messages also, so I am pushing the PR again.

Comment (sayan-biswas) [reply operator/test/test.sh]: done

**Review by gabemontero — APPROVED** [review]
/lgtm

**Pull request merged**